docs/docs/pii-management.mdx

```diff
@@ -0,0 +1,184 @@
+a pre-trained language model available, you must train your own [spaCy](https://spacy.io/usage/training), [stanza](https://stanfordnlp.github.io/stanza/training.html)
+or [huggingface](https://huggingface.co/docs/transformers/training) model and install it in your Rasa Pro environment.
+:::
+
+### How to populate the rule_lists section
+
+The `rule_lists` section contains a list of anonymization rule lists. Each rule list must have a unique `id` and a list
+of `rules`. Each rule must have an `entity` field and a `substitution` field. The `entity` field specifies the Presidio
+entity type to be anonymized and must be in uppercase.
+
+The `substitution` field specifies the anonymization method to be used. Currently, the
+following anonymization methods are supported: `text`, `mask`, and `faker`. If no substitution method is specified,
+the default substitution method is `mask`.
+
+The `value` field is only required for the `text` anonymization method. It specifies the text to be used as the anonymized value.
+If the `value` field is not specified, the original entity value to be anonymized will be replaced with a null value.
+
+The `faker` anonymization method uses the [Faker](https://faker.readthedocs.io/en/master/) library to generate fake data.
+The `faker` substitution method does not support the following Presidio entity types:
+- `CRYPTO`, `NRP`, `MEDICAL_LICENSE`
+- `US_BANK_NUMBER`, `US_DRIVER_LICENSE`
+- `UK_NHS`
+- `IT_FISCAL_CODE`, `IT_DRIVER_LICENSE`, `IT_PASSPORT`, `IT_IDENTITY_CARD`
+- `SG_NRIC_FIN`
+- `AU_ABN`, `AU_ACN`, `AU_TFN`, `AU_MEDICARE`
+
+If any of the above entities is used together with the `faker` substitution method, the anonymization pipeline will default
+to the `mask` substitution method.
+
+## How to update the Kafka event broker configuration
+
+The anonymization pipeline uses Kafka to publish the anonymized events to the Kafka topic that is mapped to the
+anonymization rule list. You can configure the Kafka event broker in the `endpoints.yml` file. The Kafka event broker
+must contain the `anonymization_topics` section, which must have the following structure:
+
+```yaml
+event_broker:
+  type: kafka
+  partition_by_sender: True
+  security_protocol: PLAINTEXT
+  url: localhost
+  anonymization_topics:
+    - name: topic_1
+      anonymization_rules: rules_1
+    - name: topic_2
+      anonymization_rules: rules_2
+  client_id: kafka-python-rasa
+```
+
+The `anonymization_topics` section contains a list of Kafka topics to which the anonymized events will be published.
+Each Kafka topic must have a `name` field and an `anonymization_rules` field. The `name` field specifies the name of the
+Kafka topic. The `anonymization_rules` field specifies the `id` of the anonymization rule list to be used for the Kafka topic.
+
+## How to enable anonymization of PII in logs
+
+You can enable anonymization of PII in logs by filling the `logger` section in the `endpoints.yml` file.
```

Comment:
How is this different from `custom-logging-configuration`?

Reply:
The logging file CLI option is used for configuring the  `logging` Python std library which was the dominant logging method we used in the codebase. However, in order to be able to hook the pipeline in PII leaking logs only (and not across all the possible logs), we chose to use `structlog` library instead which uses the [concept of processor chains](https://www.structlog.org/en/stable/getting-started.html#manipulating-log-entries-in-flight), we created our own processor (the anonymizer). In a nutshell, you can't use the logging file with structlog as far as I'm aware or it's for a separate ticket to work on how to enable this. Let me know if any of this needs to be clarified in the docs. I thought it's an implementation detail but the reality is that now the logs will be a mix of new structured logs and old logs with text messages.

Reply:
I am not sure how many are using custom logging configuration anyway, perhaps we can move custom logging configurations to rasa pro and use endpoints.yml to harmonize it. but nevertheless it is a separate conversation. I dont see it contradicting the docs here so we are good